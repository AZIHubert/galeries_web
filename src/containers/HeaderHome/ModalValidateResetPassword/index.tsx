import * as React from 'react';

import ModalContainer from '#components/ModalContainer';
import TextButton from '#components/TextButton';
import ModalTimer from '#components/ModalTimer';

import { resendResetPassword } from '#helpers/api';

interface ModalValidateResetPasswordI {
  currentEmail: string;
  loading: boolean,
  setError: React.Dispatch<React.SetStateAction<string>>;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>
}

const ModalValidateResetPassword = ({
  currentEmail,
  loading,
  setError,
  setLoading,
}: ModalValidateResetPasswordI) => {
  const [open, setOpen] = React.useState<boolean>(false);
  const handleClose = () => setOpen(false);
  const onClick = async () => {
    if (!loading) {
      setLoading(true);
      try {
        await resendResetPassword({ email: currentEmail });
        setOpen(true);
      } catch (err) {
        if (err.response) {
          if (err.status === 500) {
            setError('Something went wrong. Please try again');
          } else {
            const { errors } = err.response.data;
            setError(errors);
          }
        } else {
          setError('Something went wrong. Please try again');
        }
      }
      setLoading(false);
    }
  };
  return (
    <ModalContainer
      testId="modalValidateResetPassword"
      title='Reset your password'
      titleTextAlign='center'
    >
      <p
        data-testid='validateResetPasswordBody'
      >
        To reset your password, click the
        verification button in
        the email we sent to {currentEmail}.
        This helps keep your account secure.
      </p>
      <TextButton
        disabled={loading}
        justifyContent='center'
        onClick={onClick}
        marginBottom={30}
        marginTop={20}
        text='No email in your inbox or spam folder? Letâ€™s'
        textButton='resend it'
      />
      <ModalTimer
        handleClose={handleClose}
        open={open}
        text='email resend'
      />
    </ModalContainer>
  );
};

export default ModalValidateResetPassword;
