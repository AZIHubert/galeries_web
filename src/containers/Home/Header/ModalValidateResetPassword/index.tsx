import * as React from 'react';

import ModalContainer from '#components/ModalContainer';
import TextButton from '#components/TextButton';
import ModalTimer from '#components/ModalTimer';

interface ModalValidateResetPasswordI {
  currentEmail: string;
  loading: boolean,
  setLoading: React.Dispatch<React.SetStateAction<boolean>>
}

const ModalValidateResetPassword = ({
  currentEmail,
  loading,
  setLoading,
}: ModalValidateResetPasswordI) => {
  const [open, setOpen] = React.useState<boolean>(false);
  const handleClose = () => setOpen(false);
  return (
    <ModalContainer
      testId="modalValidateResetPassword"
      title='Reset your password'
      titleTextAlign='center'
    >
      <p
        data-testid='validateResetPasswordBody'
      >
        To reset your password, click the
        verification button in
        the email we sent to {currentEmail}.
        This helps keep your account secure.
      </p>
      <TextButton
        disabled={loading}
        justifyContent='center'
        onClick={() => {
          setLoading(true);
          setOpen(true);
        }}
        marginBottom={30}
        marginTop={20}
        text='No email in your inbox or spam folder? Letâ€™s'
        textButton='resend it'
      />
      <ModalTimer
        handleClose={handleClose}
        open={open}
        text='email resend'
      />
    </ModalContainer>
  );
};

export default ModalValidateResetPassword;
