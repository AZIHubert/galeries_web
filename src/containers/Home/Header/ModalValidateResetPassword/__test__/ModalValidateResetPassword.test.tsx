import * as React from 'react';
import renderer from 'react-test-renderer';

import ModalValidateResetPassword from '../index';

describe('ModalValidateResetPassword', () => {
  const mockedSetLoading = jest.fn;
  it('renders without crashing', () => {
    const tree = renderer.create(
      <ModalValidateResetPassword
        loading={false}
        setLoading={mockedSetLoading}
        currentEmail='mockedEmail@email.com'
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
