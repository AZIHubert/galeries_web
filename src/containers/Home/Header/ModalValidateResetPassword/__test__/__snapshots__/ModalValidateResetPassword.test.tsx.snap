// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalValidateResetPassword renders without crashing 1`] = `
<div
  data-testid="modalValidateResetPassword"
>
  <p>
    Reset your password
  </p>
  <p
    data-testid="validateResetPasswordBody"
  >
    To reset your password, click the verification button in the email we sent to 
    mockedEmail@email.com
    . This helps keep your account secure.
  </p>
  <p>
    No email in your inbox or spam folder? Letâ€™s
  </p>
  <button
    disabled={false}
    onClick={[Function]}
  >
    resend it
  </button>
  <p>
    .
  </p>
</div>
`;
